src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -1,37 +1,37 @@
 (ns puppetlabs.services.master.master-core
   (:import (java.io FileInputStream))
-  (:require [compojure.core :as compojure]
-            [me.raynes.fs :as fs]
-            [puppetlabs.puppetserver.ringutils :as ringutils]))
+  (:require [me.raynes.fs :as fs]
+            [puppetlabs.puppetserver.ringutils :as ringutils]
+            [puppetlabs.comidi :as comidi]))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Routing
 
 (defn v2_0-routes
-  "Creates the compojure routes to handle the master's '/v2.0' routes."
+  "Creates the web routes to handle the master's '/v2.0' routes."
   [request-handler]
-  (compojure/routes
-    (compojure/GET "/environments" request
+  (comidi/routes
+    (comidi/GET "/environments" request
                    (request-handler request))))
 
 (defn legacy-routes
-  "Creates the compojure routes to handle the master's 'legacy' routes
+  "Creates the web routes to handle the master's 'legacy' routes
    - ie, any route without a version in its path (eg, /v2.0/whatever) - but
    excluding the CA-related endpoints, which are handled separately by the
    CA service."
   [request-handler]
-  (compojure/routes
-    (compojure/GET "/node/*" request
+  (comidi/routes
+    (comidi/GET ["/node/" :node] request
                    (request-handler request))
-    (compojure/GET "/facts/*" request
+    (comidi/GET ["/facts/" :node] request
```

Comment:
This is a good point.  Not sure if this particular difference would be a big deal, but I think our overall approach to date has been to be as dumb about routing as possible for the master routes at the Puppet Server layer - letting core Ruby Puppet do the error handling.  Might be safer/better to just do the "*" for 'node', 'facts', and 'catalog' like is done for the other endpoints and not worry about destructuring the `:node` param that we don't appear to be using here for any of these anyway?  Hmmm ... although maybe your intent was to do it this way so that some of the metrics service stuff can make use of the destructured parameters?